hazelcast/src/main/java/com/hazelcast/spi/tenantcontrol/TenantControl.java

```diff
@@ -65,5 +64,29 @@
      * Hazelcast object from the tenant
      * This is so the TenantControl itself can be garbage collected
      */
-    void unregister();
+    void distributedObjectDestroyed();
+
+    /**
+     * Checks if tenant app is loaded at the current time and classes are available
+     *
+     * @param op passed so the tenant can filter on who is calling
+     * @return true if tenant is loaded and classes are available
+     */
+    boolean isAvailable(Operation op);
```

Comment:
Umph. We are leaking private API here and this is a big problem in terms of compatibility. `Operation` is very private API and it's class name, package, methods and implementation may change heavily between releases or even get deleted completely. Can you provide less context here?

Reply:
All I could think of is class name (String) or actual class (Class<?>) objects I'll change it to Class<?> for now as it seems more type-safe

Reply:
Again, operations might change class names and be deleted across minor releases so I'm not sure this will be good enough.

Reply:
The point of this is to filter by operation. service or distributed object names are useless in this case.

Reply:
It is very inconvenient from the standpoint of the SPI implementations as users might depend on classes, or methods, or API, which is internal and subject to change. While on this subject, I'm thinking not only what can happen in terms of when we change a class name/package, or add/remove methods but what would happen for your SPI implementation if we added a new operation like `com.hazelcast.cache.internal.SuperFastReadOperation`. How would you make sure that operation is handled properly then?

But, in an effort to proceed further with this PR, here are some questions and ideas:
• what is it that you actually use from an operation? What's important to your implementation of the SPI? The operation name? I'm thinking if we can introduce a new interface which will clearly describe the things that you require, and our private API can provide just this information. 
• alternatively, although unlikely, we can maybe think about making the tenant control part of the private API. That way, it has access to all other private API and we can change it any way we want in any release. Still, this option is rather "dirty"

Reply:
We decided not to pursue moving this SPI into a private/internal package as it wouldn't really be owned by Hazelcast and we could not then move it, remove it, augment it when and how we would like to. Instead, it would effectively be owned by Payara.

Instead, we have managed to agree on what is actually needed from the perspective of the operation. Resolving this comment as we will soon update this part of the code to avoid leaking private API.